Makefile

```diff
@@ -438,15 +415,16 @@ define manifest
 	$(call controller-gen,crd:crdVersions=v1 paths="./..." output:crd:artifacts:config=chart/k8gb/templates/)
 endef
 
-# function retrieves controller-gen path or installs controller-gen@v3.0.0 and retrieve new path in case it is not installed
 define controller-gen
-	@$(if $(filter $(CONTROLLER_GEN_PATH),$(NO_VALUE)),$(call install-controller-gen),)
+	@go install sigs.k8s.io/controller-tools/cmd/controller-gen@$(CONTROLLER_GEN_VERSION)
+	$(eval CONTROLLER_GEN_PATH = $(GOBIN)/controller-gen)
 	$(CONTROLLER_GEN_PATH) $1
 endef
 
-define install-controller-gen
-	GO111MODULE=on go get sigs.k8s.io/controller-tools/cmd/controller-gen@v0.5.0
-	$(eval CONTROLLER_GEN_PATH = $(GOBIN)/controller-gen)
+define golic
+	@go install github.com/AbsaOSS/golic@$(GOLIC_VERSION)
+	$(eval GOLIC_PATH = $(GOBIN)/golic)
+	$(GOLIC_PATH) inject $1
 endef
```

Comment:
`go build` just compiles the executable file and moves it to the destination. `go install` moves the executable file to `$GOBIN` and caches all non-main packages which are imported to `$GOPATH/pkg`. The cache will be used during the next compilation, provided the source did not change yet.

`$GOBIN` is `$GOPATH/bin` by default and `$GOPATH/bin` is attached to your `$PATH` during golang installation (although on my windows in times of GO.1.10, $GOBIN wasn't set). If customer didn't change $GOBIN, all executables installed with `go install` would be accessible via `$PATH`, so we can call directly `golic inject $1`. It wouldn't be our problem if someone change $GOBIN I guess.  

If we want to be too pedant, we can do something like this: `$(GOBIN)/golic inject $1`

Reply:
@kuritka isn't what we're doing already by assigning `$(eval GOLIC_PATH = $(GOBIN)/golic)`?:
```
$(eval GOLIC_PATH = $(GOBIN)/golic)
$(GOLIC_PATH) inject $1
```
I can just get rid of interim `$GOLIC_PATH` variable

Reply:
yes, we can run directly  `$(GOBIN)/golic ... `